src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQueryRecordToJson.java

```diff
@@ -160,7 +165,11 @@ private static void writeSimpleTypes(JsonWriter writer, String name, boolean isA
       case STRING:
         writer.value(object.toString());
         break;
-      // TODO CDAP-15342 write byte type to json writer
+      case BYTES:
+        if (object instanceof byte[]) {
```

Comment:
byte fields are also sometimes ByteBuffer, should add that instanceof check as well.

If it's neither, should throw an exception rather than silently drop the data.

Reply:
Done.